<a href="https://colab.research.google.com/github/rpizarrog/probabilidad-y-estad-stica/blob/master/Python%202023/Caso_01_Poblaci%C3%B3n_y_muestra.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Objetivo

Simular datos de una población y de una muestra describiendo la media poblacional y la media muestral para su adecuada interpretación.



# Desarrollo

Construir una población de 650 números con valores entre 18 y 30. Representando edades de personas
Determinar una muestra del 10% de la población de dichos edades.



## Cargar librerías

In [124]:
# Cargar librerías

import random
import numpy as np
import pandas as pd

random.seed(2023) # Semilla para valores aleatorios

### Generando población de edades

La variable población es una lista que contienen 650 elementos generados aleatorioamente y represnta la población de la variable de interés edad.


In [125]:
poblacion = []

for i in range(650):
    edad = random.randint(18, 30)
    poblacion.append(edad)

print(edades)

[23, 25, 28, 23, 26, 24, 19, 24, 18, 20, 25, 23, 23, 21, 20, 29, 30, 27, 30, 25, 21, 20, 26, 18, 26, 29, 27, 20, 21, 24, 21, 26, 30, 18, 19, 27, 19, 19, 28, 25, 25, 28, 18, 30, 29, 25, 18, 27, 20, 27, 25, 30, 23, 30, 21, 20, 27, 27, 30, 19, 27, 24, 22, 28, 19, 20, 27, 26, 24, 20, 20, 27, 30, 21, 26, 30, 26, 24, 26, 21, 24, 18, 18, 19, 24, 25, 25, 19, 24, 24, 23, 29, 19, 29, 23, 23, 19, 28, 18, 21, 19, 21, 21, 28, 22, 23, 18, 18, 26, 18, 23, 30, 26, 20, 20, 22, 20, 25, 21, 20, 30, 25, 22, 22, 29, 28, 26, 30, 18, 26, 24, 22, 27, 27, 28, 18, 21, 24, 19, 21, 21, 24, 26, 24, 26, 29, 21, 23, 20, 20, 28, 27, 18, 27, 24, 19, 25, 18, 20, 18, 23, 30, 23, 19, 22, 23, 28, 26, 28, 18, 23, 19, 27, 20, 23, 24, 23, 23, 28, 18, 22, 27, 19, 26, 30, 30, 23, 18, 29, 21, 22, 18, 22, 21, 19, 29, 25, 30, 19, 29, 21, 28, 30, 23, 30, 26, 25, 20, 25, 29, 23, 21, 26, 20, 30, 23, 26, 23, 18, 24, 18, 23, 22, 24, 28, 23, 28, 21, 19, 22, 30, 20, 29, 26, 29, 19, 24, 27, 30, 26, 20, 29, 23, 26, 21, 18, 20, 24, 21, 20,

Media aritmética de la población

$$
\mu = \sum_{i=1}^{N}=\frac{x_i}{N}=\frac{x_1+x_2+x_3+ ... x_{650}}{650}
$$




In [126]:
promedio = sum(poblacion) / 650
promedio

23.883076923076924

In [127]:
print(poblacion)

[24, 29, 25, 24, 23, 27, 27, 23, 28, 19, 30, 19, 28, 22, 29, 26, 29, 26, 22, 20, 21, 20, 28, 18, 21, 23, 19, 27, 20, 29, 30, 18, 21, 20, 25, 25, 26, 23, 26, 19, 19, 26, 25, 25, 18, 30, 19, 27, 21, 29, 27, 28, 29, 19, 27, 19, 19, 20, 24, 25, 28, 21, 27, 28, 20, 26, 21, 22, 22, 28, 29, 19, 27, 29, 18, 23, 25, 20, 22, 19, 26, 24, 27, 28, 23, 22, 23, 22, 20, 20, 29, 25, 19, 29, 23, 26, 21, 20, 26, 23, 26, 28, 19, 18, 22, 20, 18, 22, 20, 29, 21, 20, 28, 30, 22, 24, 26, 29, 19, 24, 30, 29, 25, 29, 24, 19, 30, 29, 28, 23, 30, 23, 19, 19, 25, 25, 25, 19, 27, 27, 24, 21, 18, 24, 27, 27, 28, 21, 23, 18, 21, 19, 19, 28, 22, 20, 22, 21, 29, 27, 28, 18, 22, 18, 24, 23, 24, 18, 29, 27, 23, 29, 22, 28, 28, 19, 29, 23, 27, 30, 26, 19, 22, 22, 18, 21, 26, 21, 23, 22, 20, 20, 26, 19, 19, 30, 21, 20, 27, 21, 28, 19, 26, 22, 27, 21, 18, 27, 27, 20, 29, 23, 25, 24, 29, 22, 29, 25, 19, 27, 21, 18, 26, 24, 20, 28, 19, 18, 30, 25, 20, 24, 27, 23, 29, 29, 28, 23, 28, 21, 27, 21, 28, 19, 18, 30, 26, 24, 19, 24,

In [128]:
# Calcular la media o el promedio utilizando NumPy y l afunción mean()
mean_edades = np.mean(poblacion)
print (mean_edades)


23.883076923076924


Describir los datos con la función describe()

Se convierte antes la lista de población a un conjunto de datos data.frame llamado df_edades.

La función describe arroja parámetros tales como la cantidd de elementos count, el valor mínimo y máximo, los cuartiles, la media y la desviación estándar.

In [129]:
df_edades = pd.DataFrame(poblacion, columns=["edades"])

# Obtener el resumen estadístico utilizando describe()
resumen_estadistico = df_edades["edades"].describe()

print(resumen_estadistico)

count    650.000000
mean      23.883077
std        3.681276
min       18.000000
25%       21.000000
50%       24.000000
75%       27.000000
max       30.000000
Name: edades, dtype: float64


### Generando una muestra del 10% a partir de datos de la población



In [130]:
muestra = random.sample(poblacion, 65)
print(muestra)

[30, 19, 22, 22, 27, 20, 28, 19, 21, 27, 29, 23, 21, 19, 23, 24, 28, 29, 28, 20, 25, 28, 23, 19, 25, 29, 26, 18, 22, 18, 26, 24, 26, 20, 29, 19, 20, 22, 27, 23, 30, 25, 19, 19, 23, 29, 22, 27, 19, 19, 26, 20, 22, 22, 23, 19, 23, 21, 27, 30, 18, 25, 20, 28, 27]


Estadísiticos de la muestra media aritmética

$$
\bar{x}=\sum_{i=1}^{n}= \frac{x_i}{n}
$$


In [131]:
n = 65
promedio_muestra = sum(muestra) / (n)
print(promedio_muestra)


23.553846153846155


In [132]:
print(mean(muestra))

23.553846153846155


### Estadísticos con describe()

Se generan los estadísticos de la muestra

In [133]:
df_muestra_edades = pd.DataFrame(muestra, columns=["edades"])

# Obtener el resumen estadístico utilizando describe()
resumen_estadistico = df_muestra_edades["edades"].describe()

print(resumen_estadistico)

count    65.000000
mean     23.553846
std       3.729237
min      18.000000
25%      20.000000
50%      23.000000
75%      27.000000
max      30.000000
Name: edades, dtype: float64



# Interpretación

Se construyó una población de 650 personas que tienen una edad entre 18 y 30 años. $N=650$

Se determinó el parámetro de la media poblacional siendo este igual 23.88. $μ = 23.88$.

Se extrajo una muestra del 10% de la población. $n=65$

La media de la muestra fue de 23.55. $\bar{x} = 23.55$

Los valores de la media son similares y la diferencia se le conoce como error muestral de la media, que es la diferencia que hay entre la media de la población y la media de la muestra.

Las diferencias encontradas con describe() entre los estadísticos de la muestra y los parámetros de la población se le llama errores muestrales.